src/cascade/input_data/configuration/construct_bundle.py

```diff
@@ -0,0 +1,113 @@
+    get_bundle_study_covariates
+from cascade.input_data.configuration.id_map import make_integrand_map
+from cascade.input_data.db.bundle import _get_bundle_id, _get_bundle_data
+from cascade.dismod.db.metadata import DensityEnum
+from cascade.core import getLoggers
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def _normalize_measures(data):
+    """Transform measure_ids into canonical measure names, for instance,
+    GBD measure 38 for birth prevalence becomes pini.
+    """
+    data = data.copy()
+    gbd_measure_id_to_integrand = make_integrand_map()
+    if any(data.measure_id == 6):
+        MATHLOG.warn(f"Found incidence, measure_id=6, in data. Should be Tincidence or Sincidence.")
+    try:
+        data["measure"] = data.measure_id.apply(lambda k: gbd_measure_id_to_integrand[k].name)
+    except KeyError as ke:
+        raise RuntimeError(
+            f"The bundle data uses measure {str(ke)} which doesn't map "
+            f"to an integrand. The map is {gbd_measure_id_to_integrand}."
+        )
+    return data
+
+
+def _normalize_sex(data):
+    """Transform sex_ids from 1, 2, 3 to male, female, both.
+    """
+    data = data.copy()
+    try:
+        data["sex"] = data.sex_id.apply({1: "Male", 2: "Female", 3: "Both", 4: "Unspecified"}.get)
+    except KeyError as ke:
+        raise InputDataError(f"Unrecognized sex id") from ke
+    return data
+
+
+def _normalize_bundle_data(data):
+    """Normalize bundle columns, strip extra columns and index on `seq`
+    """
+    data = _normalize_measures(data)
+    data = _normalize_sex(data)
+
+    data = data.set_index("seq")
+
+    cols = ["measure", "mean", "sex", "sex_id", "standard_error", "age_start", "age_end", "year_start", "year_end", "location_id"]
+
+    return data[cols]
+
+
+def bundle_to_observations(config, bundle_df):
+    """
+    Convert bundle into an internal format. It removes the sex column and changes
+    location to node. It also adjusts for the demographic specification.
+    """
+    if "location_id" in bundle_df.columns:
+        location_id = bundle_df["location_id"]
+    else:
+        location_id = np.full(len(bundle_df), config.location_id, dtype=np.int)
+
+    # assume using demographic notation because this bundle uses it.
+    demographic_interval_specification = 0
+    MATHLOG.info(f"Does this bundle assume demographic notation? {demographic_interval_specification}. "
+                 f"A 1 means that 1 year is added to both end ages and end times. A 0 means nothing is added.")
+
+    weight_method = "constant"
+    MATHLOG.info(f"The set of weights for this bundle is {weight_method}.")
+    # Stick with year_start instead of time_start because that's what's in the
+    # bundle, so it's probably what modelers use. Would be nice to pair
+    # start with finish or begin with end.
+    return pd.DataFrame(
+        {
+            "measure": bundle_df["measure"],
+            "node_id": location_id,
```

Comment:
Ok, I have seen for the USA node_id = 98, and location_id = 102; i.e. they're not different names for the same number.
So, just wondering about that here?

Reply:
It gets a node_id from its order in the Dismod file, but the location id is the GBD assignment, and those go up to 50,000.